rust/src/dcerpc/dcerpc.rs

```diff
@@ -982,40 +944,38 @@ impl DCERPCState {
         // Check if header data was complete. In case of EoF or incomplete data, wait for more
         // data else return error
         if self.bytes_consumed < DCERPC_HDR_LEN.into() && input_len > 0 {
-            parsed = self.process_header(buffer);
+            parsed = self.process_header(cur_i);
             if parsed == -1 {
-                self.extend_buffer(buffer, direction);
-                return AppLayerResult::ok();
+                return AppLayerResult::incomplete(0, DCERPC_HDR_LEN as u32);
             }
             if parsed < 0 {
                 return AppLayerResult::err();
             }
             self.bytes_consumed += parsed;
+            input_len -= parsed as usize;
         }
 
         let fraglen = self.get_hdr_fraglen().unwrap_or(0);
 
-        if (buffer.len()) < fraglen as usize {
+        if (input_len + self.bytes_consumed as usize) < fraglen as usize {
             SCLogDebug!("Possibly fragmented data, waiting for more..");
-            self.extend_buffer(buffer, direction);
-            return AppLayerResult::ok();
+            return AppLayerResult::incomplete(parsed as u32, (fraglen - parsed as u16) as u32);
```

Comment:
confusing: we use `parsed` and `fraglen` in the calculation, but `self.bytes_consumed` and `input_len` in the if statement.

Reply:
This looks like I'm creating a bug.. Thank you! Fixing